picard/tagger.py

```diff
@@ -511,81 +545,93 @@ def add_files(self, filenames, target=None):
         if new_files:
             log.debug("Adding files %r", new_files)
             new_files.sort(key=lambda x: x.filename)
-            if target is None or target is self.unclustered_files:
-                self.unclustered_files.add_files(new_files)
+            if target is self.unclustered_files:
                 target = None
+
             for file in new_files:
-                file.load(partial(self._file_loaded, target=target))
+                self._pending_files[file.filename] = (file, target)
+                self._pending_files_process_queue.put(file.filename)
 
-    def add_directory(self, path):
-        if config.setting['recursively_add_files']:
-            self._add_directory_recursive(path)
-        else:
-            self._add_directory_non_recursive(path)
+    def _file_load_finished_thread(self):
+        while True:
+            filename, metadata = self._pending_files_results_queue.get()
 
-    def _add_directory_recursive(self, path):
-        ignore_hidden = config.setting["ignore_hidden_files"]
-        walk = os.walk(path)
+            if self.stopping:
+                break
 
-        def get_files():
-            try:
-                root, dirs, files = next(walk)
-                if ignore_hidden:
-                    dirs[:] = [d for d in dirs if not is_hidden(os.path.join(root, d))]
-            except StopIteration:
-                return None
+            if filename in self._pending_files:
+                file, target = self._pending_files.pop(filename)
+
+                error = not metadata or None
+                thread.to_main(file._loading_finished, result=metadata, error=error, callback=None)
+
+                time.sleep(0)
```

Comment:
In this case, is there a difference betwen `time.sleep(0)` and https://doc.qt.io/qt-5/qthread.html#yieldCurrentThread ?